src/main/scala/net/psforever/objects/vehicles/VehicleControl.scala

```diff
@@ -327,46 +306,78 @@ class VehicleControl(vehicle: Vehicle)
         case _ =>
       }
 
-  val tryMountBehavior : Receive = {
-    case msg @ Mountable.TryMount(user, seat_num) =>
-      val exosuit = user.ExoSuit
-      val restriction = vehicle.Seats(seat_num).ArmorRestriction
-      val seatGroup = vehicle.SeatPermissionGroup(seat_num).getOrElse(AccessPermissionGroup.Passenger)
-      val permission = vehicle.PermissionGroup(seatGroup.id).getOrElse(VehicleLockState.Empire)
-      if (
-        (if (seatGroup == AccessPermissionGroup.Driver) {
-          vehicle.Owner.contains(user.GUID) || vehicle.Owner.isEmpty || permission != VehicleLockState.Locked
+  override protected def mountTest(
+                                    obj: PlanetSideServerObject with Mountable,
+                                    seatNumber: Int,
+                                    user: Player
+                                  ): Boolean = {
+    val seatGroup = vehicle.SeatPermissionGroup(seatNumber).getOrElse(AccessPermissionGroup.Passenger)
+    val permission = vehicle.PermissionGroup(seatGroup.id).getOrElse(VehicleLockState.Empire)
+    (if (seatGroup == AccessPermissionGroup.Driver) {
+      vehicle.Owner.contains(user.GUID) || vehicle.Owner.isEmpty || permission != VehicleLockState.Locked
+    } else {
+      permission != VehicleLockState.Locked
+    }) &&
+    super.mountTest(obj, seatNumber, user)
+  }
+
+  def mountCleanup(mount_point: Int, user: Player): Unit = {
+    val obj = MountableObject
+    obj.PassengerInSeat(user) match {
+      case Some(seatNumber) =>
+        //if the driver mount, change ownership if that is permissible for this vehicle
+        if (seatNumber == 0 && !obj.OwnerName.contains(user.Name) && obj.Definition.CanBeOwned.nonEmpty) {
+          //whatever vehicle was previously owned
+          vehicle.Zone.GUID(user.avatar.vehicle) match {
+            case Some(v : Vehicle) =>
+              v.Actor ! Vehicle.Ownership(None)
+            case _ =>
+              user.avatar.vehicle = None
+          }
+          LoseOwnership() //lose our current ownership
+          GainOwnership(user) //gain new ownership
```

Comment:
Does it make sense to just have `GainOwnership` call `LoseOwnership` directly? That way it can't be forgotten beforehand.

Reply:
Will be adjusted here and in `GainOwnership`.